### PR TITLE
Update/mimi js page

### PR DESCRIPTION
**Pull request by MimiDumpling**
Updating the JavaScript SDK pages with some suggestions that were recently made.

### CODE REVIEW

File: src/collections/_documentation/platforms/javascript/index.md [hunk 1]
```diff
@@ -5,12 +5,38 @@ title: JavaScript
...
+The quickest way to get started is to use the CDN hosted version of the JavaScript browser SDK:
+
+``` javascript
+<script src="https://browser.sentry-cdn.com/5.4.0/bundle.min.js" crossorigin="anonymous"></script>
+```
+
+You can also add the Sentry SDK as a dependency using yarn or npm:
+
+``` bash
+# Using yarn
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: can we only show one? people who use yarn know how to use yarn. also "npm save" is the equiv to "yarn add"

File: src/collections/_documentation/platforms/javascript/index.md [hunk 2]
```diff
@@ -5,12 +5,38 @@ title: JavaScript
...
+You should `init` the Sentry Browser SDK as soon as possible during your page load:
+
+``` javascript
+Sentry.init({ dsn: 'https://<key>@sentry.io/<project>' });
+```
+
+When using yarn or npm:
+
+``` javascript
+import * as Sentry from '@sentry/browser';
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: we could limit this to one code example, and just show the import with a comment "if you're using sentry with NPM, import it" or similar

File: src/collections/_documentation/platforms/javascript/index.md [hunk 3]
```diff
@@ -34,14 +60,7 @@ try {
...
 It's possible to throw strings as errors. In this case, the Sentry SDK will not record tracebacks.
-{%- endcapture -%}
-{%- include components/alert.html
-  title="Note"
-  content=__alert_content
-  level="warning"
-%}
 
 ### Automatically Capturing Errors
 By including and configuring the Sentry Browser SDK, Sentry will automatically attach global handlers to capture uncaught exceptions and unhandled rejections.
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: you could just ".. configuring Sentry" (the [platform] [SDK] is implied)

File: src/collections/_documentation/platforms/javascript/index.md [hunk 4]
```diff
@@ -34,14 +60,7 @@ try {
 }
 ```
 
-{% capture __alert_content -%}
 It's possible to throw strings as errors. In this case, the Sentry SDK will not record tracebacks.
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: probably not really important here

File: src/collections/_documentation/platforms/javascript/index.md [hunk 5]
```diff
@@ -345,15 +350,15 @@ If you want to keep your source maps secret and choose not to upload your source
   content=__alert_content
 %}
 
-## Context
+## Adding Context Data
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: Data is implied. "Adding Context" or "Adding Contextual Data" but I prefer the former

File: src/collections/_documentation/platforms/javascript/index.md [hunk 6]
```diff
@@ -345,15 +350,15 @@ If you want to keep your source maps secret and choose not to upload your source
...
 You can also set context when manually triggering events.
 
 ### Setting Context {#context}
 Sentry supports additional context with events. Often this context is shared among any issue captured in its lifecycle, and includes the following components:
 
 **Structured Contexts**
 
-: Specific structured contexts --- OS info, runtime information, etc.  This is typically set automatically.
+Specific structured contexts --- OS info, runtime information, etc.  This is typically set automatically.
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: does this section include snippets or is it just a list of the kind of data? if its just a list it seems like a lot of filler. if its examples I wonder if there's a way to simplify it.

File: src/collections/_documentation/platforms/javascript/index.md [hunk 7]
```diff
@@ -150,14 +169,7 @@ module.exports = {
...
-{% capture __alert_content -%}
 In case you use [SourceMapDevToolPlugin](https://webpack.js.org/plugins/source-map-dev-tool-plugin) for more fine-grained control of source map generation, leave `noSources` turned off, so Sentry can display proper source code context in event stack traces.
-{%- endcapture -%}
-{%- include components/alert.html
-  title="Note"
-  content=__alert_content
-  level="info"
-%}
 
 #### SystemJS
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: seems not applicable to 80% of javascript developers

File: src/collections/_documentation/platforms/javascript/index.md [hunk 8]
```diff
@@ -375,6 +380,17 @@ Sentry supports additional context with events. Often this context is shared amo
...
+There are two ways you can add context to events.
+
+#### Use-Case: local/event-specific
+
+[ code snippet here ]
+
+#### Use-Case: global/middleware-type context
+
+[ code snippet here ]
+
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: lets just bundle them all together

you are either:

- sending a known context (heres how)
- or sending your own custom context

we can remove "extra" context from the documentation or make it a tiny note as "setContext" is being preferred going forward

File: src/collections/_documentation/platforms/javascript/index.md [hunk 9]
```diff
@@ -728,15 +744,6 @@ For more information, see:
 - [Full documentation on User Feedback]({%- link _documentation/enriching-error-data/user-feedback.md -%})
 - [Introducing User Feedback](https://blog.sentry.io/2016/04/21/introducing-user-feedback)
 
-### Security Policy Reporting
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: :100: on removal

File: src/collections/_documentation/platforms/javascript/index.md [hunk 10]
```diff
@@ -1022,6 +1029,109 @@ basic stack trace. This exception is stored here for further data extraction.
 
 : For breadcrumbs created from HTTP requests done via the legacy `XMLHttpRequest` API. This holds the original xhr object.
 
+## Quick Reference
+This section serves as an index/quick reference for all the config options, top level methods, etc.
+
+### Structured Contexts
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: isnt this duplicated from earlier?

File: src/collections/_documentation/platforms/javascript/index.md (hunk 1 above)
Comment (untitaker) [reply src/collections/_documentation/platforms/javascript/index.md]: `npm install --save-prod` etc. But apparently `install` saves now too: https://stackoverflow.com/a/19578808/1544347

File: src/collections/_documentation/platforms/javascript/index.md [hunk 11]
```diff
@@ -5,12 +5,38 @@ title: JavaScript
 ## Integrating the SDK
 All our JavaScript-related SDKs provide the same API. Still, there are some differences between them, such as installation, which this section of the docs explains.
 
-{% include components/platform_content.html content_dir='getting-started-install' %}
+The quickest way to get started is to use the CDN hosted version of the JavaScript browser SDK:
+
+``` javascript
+<script src="https://browser.sentry-cdn.com/5.4.0/bundle.min.js" crossorigin="anonymous"></script>
```
Comment (untitaker) [inline src/collections/_documentation/platforms/javascript/index.md]: Could we use the `sdk_version` placeholder here? Seems like we should use that where we need to show a version.

See here: https://github.com/getsentry/sentry-docs/blob/236976639c47a2ea1c00e237b531dd26f24fa4d1/src/collections/_documentation/platforms/javascript/getting-started-install/browser.md

File: src/collections/_documentation/platforms/javascript/index.md (hunk 10 above)
Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: It is. Still thinking on the best way to do the Quick Reference without it being so repetitive.

File: src/collections/_documentation/platforms/javascript/index.md (hunk 11 above)
Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: Looking into this. :)

File: src/collections/_documentation/platforms/javascript/index.md (hunk 6 above)
Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: I'm wondering if we can put this information in a table or something. I think it's important to document all the types of data a dev can collect. But I also agree the way the information is presented feels like it needs a trim or a way to condense it. Earlier today, @lobsterkatie and I were discussing the table solution.

File: src/collections/_documentation/platforms/javascript/index.md (hunk 8 above)
Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: I'm okay with bundling them together as long as how it works globally is the same as how it works locally. Is that right? They work the same?

If so, will someone (@HazAT ? @untitaker ?) provide a good code snippet to go with this section?

File: src/collections/_documentation/platforms/javascript/index.md (hunk 10 above)
Comment (dcramer) [reply src/collections/_documentation/platforms/javascript/index.md]: i dont really have an opinion on whats best here, but maybe it's better to wait on building a quick ref kind of thing and get the core shipped and then add "features" if you will

File: src/collections/_documentation/platforms/javascript/index.md (hunk 8 above)
Comment (dcramer) [reply src/collections/_documentation/platforms/javascript/index.md]: They work slightly different, but to a user its "I want to add data to an event" and you can either "add context for all events that Sentry captures" or "add context to a single event" kind of thing. Will defer to other folks on code examples.

File: src/collections/_documentation/platforms/javascript/index.md (hunk 11 above)
Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: @untitaker I'm going to add this next week. Ran out of time this week. :)

File: src/collections/_documentation/platforms/javascript/index.md [hunk 12]
```diff
@@ -23,8 +40,7 @@ Then, you can see the error in your dashboard:
 [{% asset js-index/error-message.png alt="Error in Unresolved Issues with title This is my fake error message" %}]({% asset js-index/error-message.png @path %})
 
 ## Capturing Errors
-### Capturing Errors / Exceptions {#capturing-errors}
```
Comment (dcramer) [inline src/collections/_documentation/platforms/javascript/index.md]: oh one thing @EvanPurkhiser is this gonna fuck shit up?

Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: j/k Evan. It's handled. ;)

File: src/collections/_documentation/platforms/javascript/index.md [hunk 13]
```diff
@@ -5,12 +5,29 @@ title: JavaScript
...
+```
 
 ### Connecting the SDK to Sentry
 After you've completed setting up a project in Sentry, Sentry will give you a value which we call a _DSN_ or _Data Source Name_. It looks a lot like a standard URL, but it’s just a representation of the configuration required by the Sentry SDKs. It consists of a few pieces, including the protocol, public key, the server address, and the project identifier.
 
-{% include components/platform_content.html content_dir='getting-started-dsn' %}
+You should `init` the Sentry Browser SDK as soon as possible during your page load:
+
+``` javascript
+Sentry.init({ dsn: 'https://<key>@sentry.io/<project>' });
```
Comment (EvanPurkhiser) [inline src/collections/_documentation/platforms/javascript/index.md]: Wny not use the `__PUBLIC_DSN__` here?

Comment (MimiDumpling) [reply src/collections/_documentation/platforms/javascript/index.md]: @EvanPurkhiser is that recommended? All of our "Getting Started" docs have this format. I just want to make sure we're using the proven/good/recommended way.

**Pull request merged**